R/evaluate_effects.R

```diff
@@ -78,21 +78,23 @@ fSquared <- function(seminr_model, iv, dv) {
 }
 
 model_fsquares <- function(seminr_model) {
-  if (any(names(seminr_model$measurement_model) == "orthogonal_interaction")
-      | any(names(seminr_model$measurement_model) == "two_stage_interaction")
-      | any(names(seminr_model$measurement_model) == "scaled_interaction" )) {
-    return("The fSquare cannot be calculated as the model contains an interaction term and omitting either the antecedent or moderator in the interaction term will cause model estimation to fail")
-  }
-
   path_matrix <- seminr_model$path_coef
   fsquared_matrix <- path_matrix
-
   for (dv in all_endogenous(seminr_model$smMatrix)) {
-    for (iv in all_exogenous(seminr_model$smMatrix)) {
+    ifelse(length(grep("\\*",seminr_model$smMatrix[seminr_model$smMatrix[,2] == dv,1]) ) > 0,
```

Comment:
What does `seminr_model$smMatrix[seminr_model$smMatrix[,2] == dv,1]` do?  Could we factor it out as a new function for inspecting structural matrix in the file R/inspect_smMatrix.R ?

Reply:
`length(grep("\\*",seminr_model$smMatrix[seminr_model$smMatrix[,2] == dv,1]) ) > 0` could also be just called `exists_interactions(seminr_model$smMatrix)`

Reply:
And overall, see if we can write such code without manually digging through smMatrix

Reply:
```seminr_model$smMatrix[seminr_model$smMatrix[,2] == dv,1]``` is replaced by antecedents_of()

Reply:
```grep("\\*",seminr_model$smMatrix[seminr_model$smMatrix[,2] == dv,1])```  ```becomes interactions_of(dv, seminr_model$smMatrix)```

Reply:
quick suggestion: how about `all_interactions`? `interactions_of` sounds like we are going to specify a construct whose interactions we want?

Reply:
Yup, that's exactly what we are doing in this function. all_interactions is next :)

Reply:
brilliant! shoulda looked closer!